gnovm/pkg/packages/download.go

```diff
@@ -0,0 +1,83 @@
+package packages
+
+import (
+	"fmt"
+	"net/url"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/gnolang/gno/tm2/pkg/bft/rpc/client"
+	"github.com/gnolang/gno/tm2/pkg/errors"
+)
+
+func DownloadModule(pkgPath string, dst string) error {
+	modFilePath := filepath.Join(dst, ModfileName)
+	if _, err := os.Stat(modFilePath); os.IsNotExist(err) {
+		fmt.Fprintln(os.Stderr, "gno: downloading", pkgPath)
+
+		// create client from pkgpath
+		parts := strings.Split(pkgPath, "/")
+		if len(parts) < 1 {
+			return fmt.Errorf("bad pkg path %q", pkgPath)
+		}
+		rpcURL := (&url.URL{
+			Scheme: "https",
+			Host:   "rpc." + parts[0] + ":443",
+		}).String()
+		tmClient, err := client.NewHTTPClient(rpcURL)
+		if err != nil {
+			return fmt.Errorf("failed to instantiate tm2 client with remote %q: %w", rpcURL, err)
+		}
+		defer tmClient.Close()
+
+		// fetch files
+		data, err := qfile(tmClient, pkgPath)
+		if err != nil {
+			return fmt.Errorf("failed to query files list for pkg %q: %w", pkgPath, err)
+		}
+		if err := os.MkdirAll(dst, 0o744); err != nil {
+			return fmt.Errorf("failed to create cache dir for %q at %q: %w", pkgPath, dst, err)
+		}
+		files := strings.Split(string(data), "\n")
+		for _, file := range files {
+			filePath := path.Join(pkgPath, file)
+			data, err := qfile(tmClient, filePath)
+			if err != nil {
+				return fmt.Errorf("failed to query package file %q: %w", filePath, err)
+			}
+			dst := filepath.Join(dst, file)
+			if err := os.WriteFile(dst, data, 0o644); err != nil {
+				return fmt.Errorf("failed to write file at %q: %w", dst, err)
+			}
+		}
+
+		// write gno.mod
```

Comment:
do we need this?

Reply:
for example: 
if you first download a sub-module, for example `gno.land/p/demo/json/eisel_lemire`
it will create the `$GNOHOME/pkg/mod/gno.land/p/demo/json` directory since it's a parent directory
to know that the `gno.land/p/demo/json` package needs to be downloaded, I'm checking for the presence of `$GNOHOME/pkg/mod/gno.land/p/demo/json/gno.mod`